datashader/transfer_functions.py

```diff
@@ -82,22 +84,36 @@ def eq_hist(data, nbins=256):
     """
     if not isinstance(data, np.ndarray):
         raise TypeError("data must be np.ndarray")
-    if np.issubdtype(data.dtype, np.integer):
-        hist = np.bincount(data.ravel())
+    data2 = data if mask is None else data[~mask]
+    if np.issubdtype(data2.dtype, np.integer):
+        hist = np.bincount(data2.ravel())
         bin_centers = np.arange(len(hist))
         idx = np.nonzero(hist)[0][0]
         hist, bin_centers = hist[idx:], bin_centers[idx:]
     else:
-        hist, bin_edges = np.histogram(data[~np.isnan(data)], bins=nbins)
+        hist, bin_edges = np.histogram(data2, bins=nbins)
         bin_centers = (bin_edges[:-1] + bin_edges[1:]) / 2
     cdf = hist.cumsum()
     cdf = cdf / float(cdf[-1])
-    return np.interp(data.flat, bin_centers, cdf).reshape(data.shape)
+    out = np.interp(data.flat, bin_centers, cdf).reshape(data.shape)
+    return out if mask is None else np.where(mask, np.nan, out)
 
 
-_interpolate_lookup = {'log': np.log1p,
-                       'cbrt': lambda x: x ** (1/3.),
-                       'linear': lambda x: x,
+def _linear(data, mask=None):
+    return np.where(mask, np.nan, data) if mask is not None else data
+
+
+def _log(data, mask=None):
+    return np.log1p(_linear(data, mask))
+
+
+def _cbrt(data, mask=None):
+    return _linear(data, mask) ** (1/3.)
+
```

Comment:
> If that's what users should do, then we should probably just put that code repetitively into cbrt and log, so that they act as good examples for people who look to see how to write their own how options.

I guess better would be to move the implementation of `_linear` into a non-underscored (i.e., importable and reusable) helper function, e.g.:

```python
def nan_mask(data, mask=None):
    return np.where(mask, np.nan, data) if mask is not None else data

def _linear(data, mask=None):
    return nan_mask(data,mask)
```